packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -153,31 +138,47 @@ export class SOQLEditorInstance {
     }
   }
 
-  protected onDidRecieveMessageHandler(e: SoqlEditorEvent): void {
-    switch (e.type) {
+  protected onDidRecieveMessageHandler(event: SoqlEditorEvent): void {
+    switch (event.type) {
       case MessageType.UI_ACTIVATED: {
         this.updateWebview(this.document);
         break;
       }
       case MessageType.UI_SOQL_CHANGED: {
-        const soql = e.payload as string;
+        const soql = event.payload as string;
         this.updateTextDocument(this.document, soql);
         break;
       }
       case MessageType.SOBJECT_METADATA_REQUEST: {
-        this.retrieveSObject(e.payload as string).catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object metadata for the ${e.payload} object.`
-          );
-        });
+        retrieveSObject(event.payload as string)
+          .then(sobject => this.updateSObjectMetadata(sobject))
+          .catch(e => {
+            if (e.errorCode !== 'NOT_FOUND') {
+              const errorMessage =
+                `An error occurred while handling a request for object metadata for the ${event.payload} object.\n` +
+                e.message;
+
+              vscode.window.showInformationMessage(errorMessage);
+              channelService.appendLine(errorMessage);
+            }
+
+            this.updateSObjectMetadata(undefined);
+          });
         break;
       }
       case MessageType.SOBJECTS_REQUEST: {
-        this.retrieveSObjects().catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object names.`
-          );
-        });
+        retrieveSObjects()
+          .then(sobjectNames => this.updateSObjects(sobjectNames))
+          .catch(e => {
+            if (e.errorCode !== 'NOT_FOUND') {
```

Comment:
Is there a reason you are only including 'NOT_FOUND'?

Reply:
Hmm... for `SOBJECTS_REQUEST ` it probably shouldn't be done.

For the case of `SOBJECT_METADATA_REQUEST` however, what happens is that the query text might refer to a non-existent SObject. IMHO, we shouldn't show an error toast here... it's an error we should expect (say, while the user is typing a FROM clause on the editor) and should handle explicitly, for example: disabling the Fields selection dropdown, and may be show some explanation text (within the SOQL UI) that an object with that name could not be found).

Reply:
Gotcha.  I won't send 'SOBJECT_METADATA_REQUEST' to telemetry either or print it in the channel.  I added a story to handle the case in the UI.

And if they do run the query, they will see the error that the field doesn't exist anyway.  The error information coming back from run query is informative.